torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -642,6 +648,8 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
   void performNocolorSplit(at::Device device);
 
+  void extendTimeoutUntilFirstDone(const std::chrono::milliseconds& timeout);
```

Comment:
It's probably better to just come up with a term of art for it (like "ephemeral timeout") and then explain what it is in docs.

Reply:
If you call it "ephemeral", but what does it mean when we call this API multiple times which is very much likely?

Reply:
The timeout is still ephemeral, in the sense that any timeout you add via this API eventually goes away (even if you call it multiple times).